### DISCUSSION TITLE
pytest.raises(), how to access the original exception object

### DISCUSSION THREAD

**Discussion by ruck94301** [pytest-dev/pytest#8011] [Q&A] [answered]
I expect that a block will raise an exception, and I want to look at the exception's attributes.
I know to use pytest.raises(),
and AFTER the context manager, I can access some details, but not the ones I want.
```
with pytest.raises(subprocess.CalledProcessError) as pytest_e:
    result = subprocess.check_output(auditcmd)

logger.debug('%s: %r', 'pytest_e.exconly()', pytest_e.exconly())

# I want the CalledProcessError's returncode and output!
```
Can I access the original exception object stored somewhere in pytest_e?
The original exception object has a returncode attribute and an output attribute that I want access to.
If that's not possible, then my workaround is to avoid pytest.raises(), like
```
try:
    result = subprocess.check_output(auditcmd)
    assert Fail
except subprocess.CalledProcessError as genuine_e:
    assert genuine_e.returncode == 1
    result = mydecode(genuine_e.output)  # process the output string to a dictionary
    assert result['dog says'] == 'woof'
    assert result['cat says'] == 'meow'
```
Am I missing something obvious?  
Thanks for any comments!

**Comment by nicoddemus** [comment, marked as answer]
Hi @ruck94301, 

`exc_info.value` contains the original exception object:

```python
with pytest.raises(subprocess.CalledProcessError) as pytest_e:
    result = subprocess.check_output(auditcmd)

logger.debug(str(pytest_e.value))
```

See https://docs.pytest.org/en/stable/reference.html#pytest._code.ExceptionInfo.value.

**Reply by ruck94301** [reply]
right, that works for me thank you.
